### ISSUE TITLE
Add Update Gesture to Wallet

### ISSUE THREAD

**Issue by 3esmit** [status-im/status-react#10242]
# Feature Issue

It's a common feature in apps the gesture to swipe down on the panel to update it. 

Currently the wallet don't have any way of updating it by user request, relaying on a non transparent (and confusing) update methodology. 

### User Story

As a user, I want to update wallet balance so that I can be sure I have the latest information.

**Comment by hesterbruikman** [comment]
Design for this was originally specified as part of #8443. The gesture was dropped to simplify and unblock a basic loading state [here](https://github.com/status-im/status-react/issues/8443#issuecomment-553798316). PR for basic loading state [here](https://github.com/status-im/status-react/pull/10196).

Original designs (still valid). Find then below and on [Figma](https://www.figma.com/file/XUehMnhyD1FGcWzvGz6SXqvh/Wallet?node-id=4183%3A1914).

Pull-to-refresh gesture should work on:
- Wallet overview screen 
- Individual account screen

Pull-to-refresh gesture should update:
- Currency value
- Tx history

Spinner shown on pull-to-refresh should also be shown on system-initiated refresh. All existing spinners on fiat value should be removed.

<img width="761" alt="Screenshot_2020-03-30_at_15 21 48" src="https://user-images.githubusercontent.com/35233054/77929623-29d43000-72aa-11ea-855b-aa006c06e5cd.png">

> Here's an old GIF with the old design, but the concept is the same https://dribbble.com/shots/3777816-Status-Wallet

**Comment by hesterbruikman** [comment]
Further requirements on `Pull to refresh` behavior based on https://github.com/status-im/status-react/pull/11007#issuecomment-670483323

The user should always be able to trigger `Pull to refresh`, client behavior on this action depends on `mobile data setting`:
- On wifi: 
     - Fetch updates
- On mobile data:
     - `Enabled` - Fetch updates
     - `Disabled` - Prompt settings bottom sheet `Sync using mobile data`
     - `Data saver` - Fetch updates

**Comment by dotemacs** [comment]
Hello @andremedeiros 

I've started on this issue as per Ceri's instruction.

This is the direction I started to take:
https://github.com/status-im/status-react/compare/develop...dotemacs:add-update-gesture-to-wallet-10242?expand=1

To explain, I've added `refresh-control` in a simplest possible way, just show my initial approach, in order to get some feedback.

Is `quo.react-native` the right ns to pull in `RefreshControl`?
Or should I put it in some other ns?

Is the passing in of, a place holder for now, a simple refresh-control the right way to go about adding this:
https://github.com/status-im/status-react/compare/develop...dotemacs:add-update-gesture-to-wallet-10242?expand=1#diff-1ca13a9ad7da30d683f828b7bb84758fR77

This is how it looks so far (nothing mind blowing, just setting the context):
https://www.dropbox.com/s/64bs0dpham6j29p/pull-to-refresh.gif?dl=0

If the answer to the above is OK, I'll proceed further.

Thanks

**Comment by Ferossgp** [comment]
Hi @dotemacs! Yes, you are in the right direction, the `quo.react-native` is the correct ns. 
As per the second question, you can go with the default refresh-control spinner, instead of the Figma one, if I get the question right.

**Comment by dotemacs** [comment]
@Ferossgp thanks for answering my question.

**Comment by dotemacs** [comment]
Hey @hesterbruikman 


In https://github.com/status-im/status-react/pull/11102 I've added this 
![image](https://user-images.githubusercontent.com/152363/90679127-423fe980-e24f-11ea-9bcb-0fd086fb6d72.png)

When you say:

> Pull-to-refresh gesture should update:
> 
>     * Currency value
> 
>     * Tx history
> 
> 
> Spinner shown on pull-to-refresh should also be shown on system-initiated refresh. All existing spinners on fiat value should be removed.

Can you please clarify for me what you mean when you say that the currency value should be updated but spinners on fiat value should be removed?

**Comment by hesterbruikman** [comment]
Hi @dotemacs ! I get that's confusing:S. You can ignore that last sentence. It was fixed a while back. 

Individual spinners addressed an issue that still exists, but is out of the scope of this issue. Issue is that if a token is not listed on Cryptocompare updating all fiat values can fail.

**Comment by dotemacs** [comment]
@hesterbruikman thanks for the answer.

But I'm unclear if there should be spinners as shown by the arrows in the picture above.
Should I take your answer to mean **no**? So that there should be only the top spinner and no other spinners?

**Comment by hesterbruikman** [comment]
Correct, only spinner at the top. Sorry for the confusion!

cc @errorists

**Comment by dotemacs** [comment]
> Correct, only spinner at the top. Sorry for the confusion!
> 

No problem, thanks for the clarification 😄

**Comment by qoqobolo** [comment]
This behavior was found: right after installing the app assets aren't resolved in the Wallet. Existing assets do not appear in the list until they are manually added, and their actual amount is only updated after re-login. SNT is shown in the list but its amount is also updated only after re-login.

It is likely that after adding item 3 of https://github.com/status-im/status-react/pull/11102#issuecomment-701386018 (`Check address for new assets and add those not yet added manually`) this issue will be resolved.

[Status-debug-logs.zip](https://github.com/status-im/status-react/files/5318135/Status-debug-logs.zip)

![photo_2020-10-02_15-30-24](https://user-images.githubusercontent.com/67952253/94928581-338e5880-04c4-11eb-8c57-9b9151d5dc33.jpg)

**Comment by cammellos** [comment]
@johnlea-quiup could you have a quick look at this please? 
Do these designs align with the wallet updates your are planning https://www.figma.com/file/XUehMnhyD1FGcWzvGz6SXqvh/Wallet?node-id=4183%3A1914 ? 
If so, how much is the overlap, is there something we can work on that will not have to change?

Thank you

**Comment by errorists** [comment]
Just to chime in with what I've discussed with Gheorghe in DMs, the user expectation of a pull to refresh gesture is that's it's a quick action that does fetch new values for data present on the screen. We shouldn't add expensive operations that might prolong it and break that expectation. So, a refresh action should involve fetching new balance values for current tokens and their fiat conversion rate. Checking for new tokens like in this [issue](https://github.com/status-im/status-react/issues/10786), is an expensive operation and should not be included in the pull to refresh.

**Comment by John-44** [comment]
@cammellos pull to refresh works with the new wallet design updates.   

I also agree with @errorists 's point above, pull to refresh shouldn't be extended to include longer running operations.  Ideally we don't want the refresh duration to last longer than roughly 3 seconds, and quicker is better.  Longer than 3 seconds and the app will start to feel 'slow' to the user, and after a longer delay a user may start to worry that the refresh might be broken.

Now if checking for new tokens can consistently fit into the under 3 seconds refresh window (along side all the other items that need to be refreshed) then I wouldn't have an objection to including it in the pull to refresh.  

But the problem of some tokens not being shown in the wallet until they are manually added (as that described in issue #11276 ) is an important bug to fix, as Users are especially sensitive regarding anything to do with money and the perceived disappearance of money is almost guaranteed to elicit a negative emotional reaction in users!!

Does the light client in Status mobile periodically check token balances?  Could it trigger a UI refresh to include the new token type when a new type is received?   Or could we perhaps check for new tokens async, and then update the UI when we detect that a new token has been received?   Ideally the new token type should just appear when it is received, without the user having to do something like restart the app or pull to refresh.

**Comment by cammellos** [comment]
> Does the light client in Status mobile periodically check token balances? Could it trigger a UI refresh to include the new token type when a new type is received? Or could we perhaps check for new tokens async, and then update the UI when we detect that a new token has been received? Ideally the new token type should just appear when it is received, without the user having to do something like restart the app or pull to refresh.

I am not quite sure how it works now, I would have to check, but for the purpose of this task, we can just work on the UI side and trigger a reasonable action from the backend (taking into account the considerations outlined above), and we can always swap out the action at a later time I guess, to do more or less work.

**Comment by hesterbruikman** [comment]
Note that design of error state when failing to fetch data has been slightly updated.

Text below icon was added. See full design on [Figma](https://www.figma.com/file/XUehMnhyD1FGcWzvGz6SXqvh/Wallet?node-id=4183%3A1914)

![Error loading](https://user-images.githubusercontent.com/35233054/99051999-d98bb300-2598-11eb-8536-87e9e252e260.png)

**Comment by 3esmit** [comment]
Make sure that the rates from FIAT <-> CRYPTO are also updated with this gesture.

**Comment by hesterbruikman** [comment]
closing as pull to refresh was recently implemented https://github.com/status-im/status-react/pull/11478

Another issue to track moving it to the account overview screen here https://github.com/status-im/status-react/issues/11849
